# Generated by Django 5.0.6 on 2025-01-02 11:04

import django.db.models.deletion
import uuid
from django.conf import settings
from django.db import migrations, models


class Migration(migrations.Migration):

    dependencies = [
        ("audit", "0034_auto_20241211_1540"),
        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
    ]

    operations = [
        migrations.CreateModel(
            name="ErrorSeverity",
            fields=[
                (
                    "id",
                    models.UUIDField(
                        default=uuid.uuid4,
                        editable=False,
                        primary_key=True,
                        serialize=False,
                        unique=True,
                    ),
                ),
                ("code", models.CharField(blank=True, max_length=20)),
                ("description", models.CharField(blank=True, max_length=200)),
            ],
            options={
                "db_table": "CFG_ERS_ErrorSeverity",
            },
        ),
        migrations.CreateModel(
            name="ErrorLogs",
            fields=[
                (
                    "id",
                    models.UUIDField(
                        default=uuid.uuid4,
                        editable=False,
                        primary_key=True,
                        serialize=False,
                        unique=True,
                    ),
                ),
                ("created_at", models.DateTimeField(auto_now_add=True)),
                ("updated_at", models.DateTimeField(auto_now=True)),
                (
                    "deleted_at",
                    models.DateTimeField(blank=True, editable=False, null=True),
                ),
                (
                    "error_message",
                    models.TextField(
                        blank=True, db_column="erl_error_message", null=True
                    ),
                ),
                (
                    "error_type",
                    models.TextField(
                        blank=True,
                        db_column="erl_error_type",
                        max_length=128,
                        null=True,
                    ),
                ),
                (
                    "error_location",
                    models.TextField(
                        blank=True, db_column="erl_error_location", null=True
                    ),
                ),
                (
                    "user_context",
                    models.TextField(
                        blank=True, db_column="erl_user_context", null=True
                    ),
                ),
                (
                    "error_stack_trace",
                    models.TextField(
                        blank=True, db_column="erl_error_stack_trace", null=True
                    ),
                ),
                (
                    "created_by",
                    models.ForeignKey(
                        blank=True,
                        editable=False,
                        null=True,
                        on_delete=django.db.models.deletion.PROTECT,
                        related_name="+",
                        related_query_name="%(class)s_created_by",
                        to=settings.AUTH_USER_MODEL,
                    ),
                ),
                (
                    "process_log",
                    models.ForeignKey(
                        blank=True,
                        db_column="erl_process_log_id",
                        null=True,
                        on_delete=django.db.models.deletion.PROTECT,
                        related_name="error_log_process_log",
                        to="audit.processloghdr",
                    ),
                ),
                (
                    "updated_by",
                    models.ForeignKey(
                        blank=True,
                        editable=False,
                        null=True,
                        on_delete=django.db.models.deletion.PROTECT,
                        related_name="+",
                        related_query_name="%(class)s_updated_by",
                        to=settings.AUTH_USER_MODEL,
                    ),
                ),
                (
                    "error_severity",
                    models.ForeignKey(
                        blank=True,
                        db_column="erl_error_severity",
                        null=True,
                        on_delete=django.db.models.deletion.PROTECT,
                        related_name="error_log_error_severity",
                        to="audit.errorseverity",
                    ),
                ),
            ],
            options={
                "db_table": "OPT_ERL_ErrorLogs",
            },
        ),
    ]
